### PR TITLE
py-google-...: add new versions and few new packages

### PR DESCRIPTION
**Pull request by w8jcik**
Needed by `googlebatch` plugin of Snakemake 8.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-google-api-core/package.py [hunk 1]
```diff
@@ -9,25 +9,44 @@
...
         depends_on("py-googleapis-common-protos@1.56.2:1", type=("build", "run"))
         depends_on("py-protobuf@3.19.5:3.19,3.20.2:4.20,4.21.6:4", type=("build", "run"))
         depends_on("py-google-auth@2.14.1:2", type=("build", "run"))
         depends_on("py-requests@2.18:2", type=("build", "run"))
 
+        with when("+grpc"):
+            depends_on("py-grpcio-status@1.49.1:1", when="^python@3.11:")
+            depends_on("py-grpcio-status@1.33.2:1", when="@2.2.0:")
+            depends_on("py-grpcio@1.49.1:1", when="^python@3.11:")
+            depends_on("py-grpcio@1.33.2:1")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-google-api-core/package.py]: These all need `type="run"`, same below

File: var/spack/repos/builtin/packages/py-google-cloud-storage/package.py [hunk 2]
```diff
@@ -9,22 +9,32 @@
...
 
     with when("@2:"):
+        depends_on("py-google-auth@2.23.3:2", type=("build", "run"), when="@2.12:")
         depends_on("py-google-auth@1.25:2", type=("build", "run"))
         depends_on("py-google-api-core@1.31.5:1,2.3.1:2", type=("build", "run"))
         depends_on("py-google-cloud-core@2.3:2", type=("build", "run"))
+        depends_on("py-google-resumable-media@2.6:", type=("build", "run"), when="@2.11:")
         depends_on("py-google-resumable-media@2.3.2:", type=("build", "run"))
         depends_on("py-requests@2.18:2", type=("build", "run"))
+        depends_on("py-google-crc32c", type=("build", "run"), when="@2.12:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-google-cloud-storage/package.py]:
```suggestion
        depends_on("py-google-crc32c@1", type=("build", "run"), when="@2.12:")
```

File: var/spack/repos/builtin/packages/py-google-resumable-media/package.py [hunk 3]
```diff
@@ -14,6 +14,7 @@ class PyGoogleResumableMedia(PythonPackage):
 
     license("Apache-2.0")
 
+    version("2.7.0", sha256="5f18f5fa9836f4b083162064a1c2c98c17239bfda9ca50ad970ccf905f3e625b")
     version("2.4.1", sha256="15b8a2e75df42dc6502d1306db0bce2647ba6013f9cd03b6e17368c0886ee90a")
     version("0.3.2", sha256="3e38923493ca0d7de0ad91c31acfefc393c78586db89364e91cb4f11990e51ba")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-google-resumable-media/package.py:20]: google-crc32c now has an upper bound

File: var/spack/repos/builtin/packages/py-grpcio-status/package.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+class PyGrpcioStatus(PythonPackage):
+    """Status proto mapping for gRPC."""
+
+    homepage = "https://grpc.io/"
+    pypi = "grpcio_status/grpcio-status-1.52.0.tar.gz"
+
+    license("Apache-2.0")
+
+    version("1.52.0", sha256="602a3808d485a1b69e11d150a075e3e927f647f3a8b387ee3c3f01633445b2fc")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-grpcio-status/package.py]: This release was yanked, I don't think we should add this version: https://github.com/grpc/grpc/issues/32306

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-grpcio-status/package.py (hunk 4 above)
Comment (w8jcik) [reply var/spack/repos/builtin/packages/py-grpcio-status/package.py]: Replaced with newer version.

File: var/spack/repos/builtin/packages/py-google-resumable-media/package.py (hunk 3 above)
Comment (w8jcik) [reply var/spack/repos/builtin/packages/py-google-resumable-media/package.py:20]: Added upper bound

File: var/spack/repos/builtin/packages/py-google-api-core/package.py (hunk 1 above)
Comment (w8jcik) [reply var/spack/repos/builtin/packages/py-google-api-core/package.py]: Added

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**